code_conventions.md

```diff
@@ -172,156 +125,46 @@ for (int i = 0; i < len; i++)
-~~~~~~~~~~~~~~~
-
-
-Null pointer (C++)
-----
-For C++, never use `NULL`, but use `nullptr` instead. NULL is an integer, not a pointer.
+```
 
-Ordering
-----
-* Members: Implement functions Top-Down, starting with constructors/deconstructors in case of classes. That way, a class implementation can be read as a page from a book: from top to bottom providing clarity. This can be an issue for declarative language constructs like C/C++ , but then it's good practice to use forward declarations.
-* Go from public, protected to private. Reasoning behind this is similar. When using a class, one is more interested in the public items. For inheriting the protected ones can be interesting, while the private parts should only be meaningfull to the maintainer of the class. This helps with the OOP paradigm of implementation hiding.
+## Ordering
+Functions SHOULD be implemented Top-Down, starting with constructors/deconstructors (in case of classes). That way, a class implementation can be read as a page from a book: from top to bottom. This can be an issue for declarative language constructs like C/C++ , but it's good practice to use forward declarations in the first place.
 
-Strings
-----
-Strings are double quotes. While python and php allow single and double quoted strings, and PEP8 only says "pick a rule and stick to it". We decided to do double quotes to match C++.
+Go from public, protected to private. When using a class, one is usually more interested in the public items. For inheriting the protected ones can be interesting, while the private parts should only be meaningful to the maintainer of the class itself.
 
-Lines and Linelength
-----
-* "Maximum line length" - There is no hard limit on the line length, but as a thumb of rule, try to keep it to at most 120 characters. This helps doing the code reviews!
-* A single line contains only one statement
+## Strings
+Strings MUST be defined by using double quotes. While Python and PHP allow single and double quoted strings, we decided to do double quotes in order to match with C++.
 
-Alignment
-----
-Align code for better readability.
-This can be done on assignment level, parameter passing on function, array definitions and so on. 
+## Lines and length
+There is no hard limit on the line length, but as a thumb of rule, try to keep it to at most 120 characters.
 
+A single line SHOULD only contain one statement.
 
-Code Guidelines
-====
-Below are a couple of guidelines which should be followed, unless there's good reason not to.
+## Class Files
+Each class SHOULD have its own file with a filename corresponding to the class name.
 
-Implementation (C/C++)
-----
-All implementations should be in .cpp files. An exception are template functions, which must be implemented in the header file.
+Hence a class named Printer would have a printer.h + printer.cpp for C/C++, a printer.py for Python and a printer.php for PHP.
```

Comment:
I'm missing the casing conventions in filenames here.

Reply:
Because we never wrote them. I think we should add them, but this was more intended as cleanup than "Lets do all the codestyle changes now".